integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/management/CarbonLoadDataCommand.scala

```diff
@@ -209,16 +209,24 @@ case class CarbonLoadDataCommand(
     * 4. Session property CARBON_OPTIONS_SORT_SCOPE
     * 5. Default Sort Scope LOAD_SORT_SCOPE
     */
-    optionsFinal.put("sort_scope",
-      options.getOrElse("sort_scope",
-        carbonProperty.getProperty(
-          CarbonLoadOptionConstants.CARBON_TABLE_LOAD_SORT_SCOPE + table.getDatabaseName + "." +
-          table.getTableName,
-          tableProperties.asScala.getOrElse("sort_scope",
-            carbonProperty.getProperty(CarbonLoadOptionConstants.CARBON_OPTIONS_SORT_SCOPE,
-              carbonProperty.getProperty(CarbonCommonConstants.LOAD_SORT_SCOPE,
-                CarbonCommonConstants.LOAD_SORT_SCOPE_DEFAULT))))))
-
+    if (tableProperties.get("sort_columns") != null &&
+        tableProperties.get("sort_scope") == null) {
+      // If there are Sort Columns given for the table and Sort Scope is not specified,
+      // we will take it as whichever sort scope given or LOCAL_SORT as default
+      optionsFinal.put("sort_scope", carbonProperty.getProperty("carbon.load.sort.scope",
+        SortScopeOptions.getSortScope("LOCAL_SORT").toString))
```

Comment:
In carbonTableSchemaCommon.scala, same logic there.

see TableNewProcessor. process(). look up for log " // If sort_scope is not specified, but sort_columns are present, set sort_scope as"

This logic doesn't handle upgrade case as table is not created again. so remove that logic and test agin

Reply:
Removed that redundant logic.